### PR TITLE
Fix #55678

### PR DESCRIPTION
**Pull request by mukeshpanchal27**
Trac ticket: https://core.trac.wordpress.org/ticket/55678

---
**This Pull Request is for code review only. Please keep all other discussion in the Trac ticket. Do not merge this Pull Request. See [GitHub Pull Requests for Code Review](https://make.wordpress.org/core/handbook/contribute/git/github-pull-requests-for-code-review/) in the Core Handbook for more details.**

### CODE REVIEW

File: src/wp-admin/includes/class-wp-filesystem-direct.php [hunk 1]
```diff
@@ -484,10 +484,10 @@ public function mtime( $file ) {
 	 * @since 2.5.0
 	 *
 	 * @param string $file Path to file.
-	 * @return int|false Size of the file in bytes on success, false on failure.
+	 * @return int Size of the file in bytes, 0 otherwise.
 	 */
 	public function size( $file ) {
-		return @filesize( $file );
+		return is_readable( $file ) ? (int) filesize( $file ) : 0;
```
Comment (spacedmonkey) [inline src/wp-admin/includes/class-wp-filesystem-direct.php]:
```suggestion
		return wp_filesize( $file );
```

This seems cleaner.

**Review by spacedmonkey — CHANGES_REQUESTED** [review]
Let's use wp_filesize

File: src/wp-admin/includes/class-wp-filesystem-direct.php (hunk 1 above)
Comment (mukeshpanchal27) [reply src/wp-admin/includes/class-wp-filesystem-direct.php]: Thanks, @spacedmonkey for the feedback. I have updated the code as per your suggestion.

**Comment by sybrew** [comment]
This change will add two filters to the class, which it and its parent previously forwent entirely (aside from deprecation handling). Please reconsider.

**Comment by spacedmonkey** [comment]
> Please reconsider.

Why? What is the issue with adding filters here?

**Comment by spacedmonkey** [comment]
Doesn't this ticket make sense to do as part of - https://core.trac.wordpress.org/ticket/55376

**Comment by sybrew** [comment]
I understand. This pull is a duplicate of https://github.com/WordPress/wordpress-develop/pull/2402.

**Pull request closed without merging**